Update claude-pr-review caller — pass App secrets via inherit (verify APP_ID/APP_PRIVATE_KEY exist first)

## Symptom

`.github/workflows/claude-pr-review.yml` will be in `startup_failure` state on the next PR. Last successful run was 2026-04-29 (no PRs since), but glitchwerks/github-actions v2.6.x rolled out a caller-contract change that this repo's caller does not satisfy.

## Root cause

The reusable workflow `glitchwerks/github-actions/.github/workflows/claude-pr-review.yml@v2` (now `v2.6.1`) added two `required: true` workflow_call secrets in glitchwerks/github-actions#252:

- `app_id`
- `app_private_key`

Our caller currently only forwards `claude_code_oauth_token`, so the App-token-generation step inside the action runs with empty inputs and fails:

```
##[error]No authentication token provided. Set app_id + app_private_key inputs.
```

## ⚠️ Pre-fix verification needed

API enumeration of this repo's Actions secrets returned **only** `CLAUDE_CODE_OAUTH_TOKEN` and `GH_TOKEN`:

```
$ gh api repos/cbeaulieu-gt/job-matcher/actions/secrets
total_count: 2
- CLAUDE_CODE_OAUTH_TOKEN
- GH_TOKEN
```

Environments (`dev`, `github-pages`) have no secrets either.

**Before merging the fix PR**, please verify in the GitHub UI (Settings → Secrets and variables → Actions) whether `APP_ID` and `APP_PRIVATE_KEY` are actually present. If they aren't, add them — without them the `secrets: inherit` fix won't resolve anything; CI will still fail.

If the secrets are missing and you don't want to configure them, alternatives are:
- Pin this repo back to `@v2.5.1` (loses the v2.6.1 severity-grep fix)
- Wait for an upstream change to make App secrets optional (would need a separate issue in glitchwerks/github-actions)

Same caveat applies to `tag-claude.yml` and `lint-failure.yml` — both reference `${{ secrets.APP_ID }}` and `${{ secrets.APP_PRIVATE_KEY }}` directly, which evaluate to empty if the secrets don't exist.

## Fix (assuming secrets are present)

Update `.github/workflows/claude-pr-review.yml`:

```yaml
jobs:
  review:
    uses: glitchwerks/github-actions/.github/workflows/claude-pr-review.yml@v2
    secrets: inherit
```

(No `authorized_users` needed — cbeaulieu-gt is the repo owner, so `author_association` resolves to `OWNER`.)

## Acceptance criteria

- [ ] `APP_ID` and `APP_PRIVATE_KEY` confirmed present in repo secrets (manual UI verification)
- [ ] `.github/workflows/claude-pr-review.yml` uses `secrets: inherit`
- [ ] A new PR shows the `claude-pr-review` job running to completion (not `startup_failure`)
- [ ] PR closes this issue

## References

- Triggering merges: glitchwerks/mom-bot#23, glitchwerks/siege-web#327 (same fix shape)
- Upstream caller-contract change: glitchwerks/github-actions#252
- Prevention follow-up: glitchwerks/github-actions#260

🤖 _Generated by Claude Code on behalf of @cbeaulieu-gt_

--- a/.github/workflows/claude-pr-review.yml
+++ b/.github/workflows/claude-pr-review.yml
@@ -12,5 +12,4 @@ jobs:
       issues: write
       pull-requests: write
       statuses: write
-    secrets:
-      claude_code_oauth_token: ${{ secrets.CLAUDE_CODE_OAUTH_TOKEN }}
+    secrets: inherit

